server/src/main/java/com/objectcomputing/checkins/services/memberprofile/MemberProfileServicesImpl.java

```diff
@@ -237,6 +217,47 @@ public List<MemberProfile> getSubordinatesForId(UUID id) {
     @Override
     @CacheInvalidate(cacheNames = {"member-cache"})
     public MemberProfile updateProfile(MemberProfile memberProfile) {
+        if (memberProfile.getId() == null) {
+            throw new BadArgException("Null profile id in update");
+        }
+
+        MemberProfile currentUser = currentUserServices.getCurrentUser();
+        boolean isAdmin = currentUserServices.isAdmin();
+        if (!isAdmin && (currentUser == null || !currentUser.getId().equals(memberProfile.getId()))) {
```

Comment:
Was this an intentional or accidental can kicking? 🤣

Reply:
Well, intentional in that there are no required permissions to update a member profile.  Since every member can update their own profile, I didn't see a way to limit which profile could be updated except by using the existing mechanism of checking for admin.  But, as I'm writing this, I'm thinking that we could add a permission to be able to "update all profiles" and specifically check for it here instead of admin.  Thoughts?

Reply:
Yes, I think that would work!

Reply:
👍 I'm onboard with that.